cmake/CMakeLists.txt

```diff
@@ -734,16 +734,23 @@ if(PKG_KIM)
   option(DOWNLOAD_KIM "Download KIM-API from OpenKIM instead of using an already installed one" ${DOWNLOAD_KIM_DEFAULT})
   if(DOWNLOAD_KIM)
     message(STATUS "KIM-API download requested - we will build our own")
-    enable_language(C)
-    enable_language(Fortran)
+    if("$ENV{CC}" STREQUAL "")
```

Comment:
@ellio167 i am a bit confused by these changes. This establishes a different way to communicate which compilers to use to the KIM-API build and it would not pass the choice given on the cmake command line to the library build. 
https://lammps.sandia.gov/doc/Build_basics.html#compile

Reply:
That is not very cmakeish. And isn't `kim_build`'s CMake already picking up `FC` and `CC` anyhow? There would only be a different if a user change the environment between running `cmake` and `make` for LAMMPS.

Reply:
Hi All,

@akohlmey Agreed; my solution needs update to honor the CMAKE_C_COMPILER and CMAKE_Fortran_COMPILER flags.

@junghans The problem is that if I set the environment only on the lammps configure command it is not propagated to the kim api build.  I would have to do something like this:

```bash
% cd lammps
% mkdir build
% cd build
% FC=gfortran-8 cmake ../cmake -DPKG_KIM=ON -DDOWNLOAD_KIM=ON
% FC=gfortran-8 make
```
I would like to not have to give the `FC=gfortran-8` on the `make` command

Reply:
> @junghans The problem is that if I set the environment only on the lammps configure command it is not propagated to the kim api build. I would have to do something like this:
> 
> ```shell
> % cd lammps
> % mkdir build
> % cd build
> % FC=gfortran-8 cmake ../cmake -DPKG_KIM=ON -DDOWNLOAD_KIM=ON
> % FC=gfortran-8 make
> ```
> 
> I would like to not have to give the `FC=gfortran-8` on the `make` command

@ellio167 i just tried this on my machine with the current master branch. setting a custom fortran compiler via the FC environment **is** propagated to the KIM cmake setup. This is with cmake 3.14.3 on fedora 29.

Reply:
@akohlmey, Ah, you are right.  The change in this PR, removes the `enable_language(Fortran)` line and then, in that case, the FC environment is not propagated...

Since the KIM library is (only) a shared library, the lammps build should not need to enable Fortran.  The kim library is already linked to the fortran library and lammps just needs to link to kim, not "kim and fortran".

Reply:
true, but since we apparently cannot have our cake and eat it, i'd prefer a spurious link, that will do no harm and has to be present at run time anyway for the shared object to load, over inconsistent compilation and complex hacks.

Reply:
on top of that, the *recommended* way to use the KIM-API is to install it properly, so we are really looking at a rather uncommon corner case here.

Reply:
agreed

Reply:
ok. then i am closing this PR now, but will leave issue #1439 open, so we can review the procedure based on your suggestions after the stable release. further discussions should be added there as well.